numpy/_array_api/_manipulation_functions.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from ._array_object import ndarray
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._types import Optional, Tuple, Union, array
+
+import numpy as np
+
+# Note: the function name is different here
+def concat(arrays: Tuple[array, ...], /, *, axis: Optional[int] = 0) -> array:
+    """
+    Array API compatible wrapper for :py:func:`np.concatenate <numpy.concatenate>`.
+
+    See its docstring for more information.
+    """
+    arrays = tuple(a._array for a in arrays)
```

Comment:
Can you give an example where this is wrong?

Reply:
```
np.concatenate([np.arange(5, dtype=np.float32), np.array(3.)], axis=None)
```
Of course, you may not actually support `axis=None` (in which case it should raise an error) and the only problem would be that you do allow type promotions that `result_type` does not normally allow.

Reply:
Thanks. Can the same issue occur with `stack`? It doesn't have a dtype argument.

Reply:
> It doesn't have a dtype argument.

Hmmpf, I just added it to concat for strings (which didn't happen yet).  You could add it for stack, it likely doesn't matter there, since it has no `axis=None` and 0-D input probably is made 1-D anyway first, but dunno.  If you need it, you will have to add it (which should be just passing it on to concatenate)